app/modules/knights.py

```diff
@@ -0,0 +1,28 @@
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.armour = knight["armour"]
+        self.weapon = knight["weapon"]
+        self.potion = knight["potion"]
+        self.protection = 0
+
+        # add "protection" stats from "armor" items
+        for a in self.armour:
+            self.protection += a["protection"]
+
+        # add "power" stats from "weapon" items
+        for w in self.weapon:
+            self.power += w[self.weapon]
```

Comment:
It is just one item, not list of items
self.weapon - dict. Just do:
self.power += self.weapon["power"]